src/Microsoft.DotNet.Interactive/Commands/KernelCommand.cs

```diff
@@ -47,5 +60,10 @@ public virtual async Task InvokeAsync(KernelInvocationContext context)
 
             await Handler(this, context);
         }
+
+        internal void SuppressExecution()
```

Comment:
Why take this approach instead of just not generating the commands in the first place?

Reply:
we do need kernels choosing commands at least, right?

Reply:
will try that way

Reply:
Yes. I think we can pretty safely filter based on command types.

Reply:
Much cleaner now.